Framework/PythonInterface/plugins/algorithms/IntegratePeaksShoeboxTOF.py

```diff
@@ -0,0 +1,687 @@
+            issues["PeaksWorkspace"] = "PeaksWorkspace must have same instrument as the InputWorkspace."
+        if pk_ws.getNumberPeaks() < 1:
+            issues["PeaksWorkspace"] = "PeaksWorkspace must have at least 1 peak."
+        # check that is getting dTOF from back-to-back params then they are present in instrument
+        if self.getProperty("GetNBinsFromBackToBackParams").value:
+            # check at least first peak in workspace has back to back params
+            if not inst.getComponentByName(pk_ws.column("BankName")[0]).hasParameter("B"):
+                issues[
+                    "GetNBinsFromBackToBackParams"
+                ] = "Workspace doesn't have back to back exponential coefficients defined in the parameters.xml file."
+        return issues
+
+    def PyExec(self):
+        # get input
+        ws = self.getProperty("InputWorkspace").value
+        peaks = self.getProperty("PeaksWorkspace").value
+        # shoebox dimensions
+        get_nbins_from_b2bexp_params = self.getProperty("GetNBinsFromBackToBackParams").value
+        nfwhm = self.getProperty("NFWHM").value
+        nshoebox = self.getProperty("NShoeboxInWindow").value
+        # shoebox optimisation
+        do_optimise_shoebox = self.getProperty("OptimiseShoebox").value
+        weak_peak_strategy = self.getProperty("WeakPeakStrategy").value
+        weak_peak_threshold = self.getProperty("WeakPeakThreshold").value
+        # validation
+        integrate_on_edge = self.getProperty("IntegrateIfOnEdge").value
+        nrows_edge = self.getProperty("NRowsEdge").value
+        ncols_edge = self.getProperty("NColsEdge").value
+        # corrections
+        do_lorz_cor = self.getProperty("LorentzCorrection").value
+        # saving file
+        output_file = self.getProperty("OutputFile").value
+
+        # create output table workspace
+        peaks = self.exec_child_alg("CloneWorkspace", InputWorkspace=peaks, OutputWorkspace="out_peaks")
+
+        array_converter = InstrumentArrayConverter(ws)
+        weak_peaks_list = []
+        ipks_strong = []
+        results = np.full(peaks.getNumberPeaks(), None)
+        prog_reporter = Progress(self, start=0.0, end=1.0, nreports=peaks.getNumberPeaks())
+        for ipk, peak in enumerate(peaks):
+            prog_reporter.report("Integrating")
+            status = PEAK_STATUS.NO_PEAK
+            intens, sigma = 0.0, 0.0
+
+            detid = peak.getDetectorID()
+            bank_name = peaks.column("BankName")[ipk]
+            pk_tof = peak.getTOF()
+
+            # check TOF is in limits of x-axis
+            xdim = ws.getXDimension()
+            if xdim.getMinimum() < pk_tof < xdim.getMaximum():
+                # get shoebox kernel for initial integration
+                nrows = self.getProperty("NRows").value  # get these inside loop as overwritten if shoebox optimised
+                ncols = self.getProperty("NCols").value
+                ispec = ws.getIndicesFromDetectorIDs([detid])[0]
```

Comment:
Is it possible to rename `ispec` to perhaps `spec_number`?

Reply:
So `ispec` refers to spectrum index, in mantid we also have spectrum number which is general is different! Do you still want me to rename? Perhaps you prefer `spec_index`? Just makes for quite long line lengths sometimes!

Reply:
Ok. Makes sense. It's fine to leave at it is.